agent/agent.go

```diff
@@ -83,12 +85,13 @@ type Options struct {
 }
 
 type Client interface {
-	Metadata(ctx context.Context) (agentsdk.Metadata, error)
+	Manifest(ctx context.Context) (agentsdk.Manifest, error)
 	Listen(ctx context.Context) (net.Conn, error)
 	ReportStats(ctx context.Context, log slog.Logger, statsChan <-chan *agentsdk.Stats, setInterval func(time.Duration)) (io.Closer, error)
 	PostLifecycle(ctx context.Context, state agentsdk.PostLifecycleRequest) error
 	PostAppHealth(ctx context.Context, req agentsdk.PostAppHealthsRequest) error
 	PostStartup(ctx context.Context, req agentsdk.PostStartupRequest) error
+	PostMetadata(ctx context.Context, key string, req agentsdk.PostMetadataRequest) error
```

Comment:
If someone has a lot of metadata, I could see us wanting to debounce some of the requests... e.g. 5 things polling each second * 1000 workspaces = a *lot* of requests

Reply:
I considered this but didn't want to pursue optimizing it immediately because the feature is very optional and there's a clear path to adding a WebSocket endpoint to improve performance when the time comes.

Reply:
A WebSocket endpoint would add substantial complexity (making this harder to review...) and bug risk.